### PR TITLE
Reverse proxy implementation for arcticwolf

### PR DESCRIPTION
**Pull request by swapnil-harness**
Implementation of reverse-proxy to support write-target on ALB for arcticwolf

### CODE REVIEW

File: src/harness/reverse-proxy/main.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import (
+	"log"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+)
+
+func main() {
+	target := "BACKEND_SERVICE_URL"
```
Comment (swapnil-harness) [inline src/harness/reverse-proxy/main.go:12]: Need to replace with backend_service_url as target

File: src/harness/reverse-proxy/main.go [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	proxyURL, _ := url.Parse(target)
+
+	proxy := httputil.NewSingleHostReverseProxy(proxyURL)
+	proxy.Director = func(req *http.Request) {
+		req.URL.Scheme = proxyURL.Scheme
+		req.URL.Host = proxyURL.Host
+
+		//rewrite logic here.
+		req.URL.Path = strings.Replace(req.URL.Path, "<old-path>", "<new-path>", 1)
```
Comment (swapnil-harness) [inline src/harness/reverse-proxy/main.go:21]: Update path here

File: src/harness/reverse-proxy/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,12 @@
+FROM golang:1.16-alpine as builder
+
+WORKDIR /app
+COPY . .
+RUN go mod init https://github.com/harness/helm-charts/src/harness/reverse-proxy
```
Comment (swapnil-harness) [inline src/harness/reverse-proxy/Dockerfile:5]: Not sure if this needs tobe modified to reflect branch https://github.com/harness/helm-charts/tree/main/src/harness/reverse-proxy

Comment (swapnil-harness) [reply src/harness/reverse-proxy/Dockerfile:5]: or move reverse-proxy folder under "helm-charts" dir

**Comment by arya-harness** [comment]
@swapnil-harness is it okay to close this PR?

**Pull request closed without merging**